src/arkode/arkode.c

```diff
@@ -2820,6 +2813,7 @@ int arkPredict_Bootstrap(ARKodeMem ark_mem, realtype hj,
 
   /* call fused vector operation to compute prediction */
   retval = N_VLinearCombination(nvec+2, cvals, Xvecs, yguess);
+  SUNCheckCallNoRet(retval);
   if (retval != 0)  return(ARK_VECTOROP_ERR);
   return(ARK_SUCCESS);
```

Comment:
I don't understand why this effectively checks the same error twice: once inside `SUNCheckCallNoRet` and then again manually.  Can we just remove the third line, since any error would have already been handled?  We could even combine both of the first two, e.g.,
```suggestion
  SUNCheckCallNoRet(N_VLinearCombination(nvec+2, cvals, Xvecs, yguess));
  return(ARK_SUCCESS);
```

Reply:
No, the integrator error handlers are not being removed because are not changing their return types to `SUNErrCode` and it would break semantics in my opinion to return `SUNErrCode` from a function that returns an `int`. We check the error with the macro so that the handlers are invoked, but we let the existing integrator logic decide what to do with the control flow.

Reply:
Also, while we could eliminate the third line in theory, it would require significant work to ensure that we do not disrupt control flow because the `SUNCheckCall` macro returns only if there was an error, not on success.